accepted/2020/binary-data/binary-data.md

```diff
@@ -0,0 +1,151 @@
+  providing overloads with the existing low-level concepts, such as span.
+
+## Design
+
+We're concerned with hard coding the type to `System.Text.Json`, but we
+acknowledge that defaults are very valuable. If we need to support other
+serializers we can provide additional arguments or APIs.
+
+This type can't live in corlib because it would depend on `System.Text.Json`
+(and whatever future serializers it needs to support). Hence, most areas in the
+BCL wouldn't be able to take `BinaryData`, but technologies above the BCL could
+(Microsoft.Extensions.*, ASP.NET Core, Azure SDK etc). While that might be
+unfortunate, we believe it's acceptable because most of the usability gains
+would be in higher levels anyway.
+
+The type isn't meant to unify different representations. Rather, it's a helper
+to make it easier for users to convert data. Thus, it's not meant as
+high-performance exchange type, like span (it is implicitly convertible to
+`ReadOnlyMemory<byte>` and `ReadOnlySpan<byte>`).
+
+The type is for "user data", that is, the contract for the API is "give me a
+blob of binary data". It's not for protocol-like APIs where the underlying
+encoding is part of the API contract, e.g. a REST API.
+
+We don't see the need to, for example, provide new virtual methods on Stream or
+`HttpClient`.
+
+Please note: Neither the constructors nor the factories copy data -- they wrap
+the input data, so long it's already binary. Anything that requires
+transcoding/serialization (e.g. string, object) isn't wrapped. While the goal
+isn't to be high performant, the goal is still to not be wasteful. Idiomatic use
+of the type shouldn't inflict copies.
+
+### Shipping vehicle
+
+**Assembly**: System.BinaryData.dll
+
+**NuGet Package**: System.BinaryData (stable, targeting `netstandard2.0`)
+
+### API
+
+```C#
+namespace System
+{
+    public class BinaryData
+    {
+        public BinaryData(byte[] data);
+        public BinaryData(object jsonSerializable, JsonSerializerOptions options = default, Type? type = null);
+        public BinaryData(ReadOnlyMemory<byte> data);
+        public BinaryData(string data);
+        public static BinaryData FromBytes(ReadOnlyMemory<byte> data);
+        public static BinaryData FromBytes(byte[] data);
+        public static BinaryData FromObjectAsJson<T>(T jsonSerializable, JsonSerializerOptions options = default, CancellationToken cancellationToken = default);
+        public static BinaryData FromStream(Stream stream);
+        public static Task<BinaryData> FromStreamAsync(Stream stream, CancellationToken cancellationToken = default);
+        public static BinaryData FromString(string data);
+        public static implicit operator ReadOnlyMemory<byte>(BinaryData data);
+        public static implicit operator ReadOnlySpan<byte>(BinaryData data);
+        public ReadOnlyMemory<byte> AsBytes();
+        public T ToObjectFromJson<T>(JsonSerializerOptions options = default, CancellationToken cancellationToken = default);
```

Comment:
Is it correct that this (and FromObjectAsJson) should take cancellationToken?
The sync version of JsonSerializer.Serialize and JsonSerializer.Deserialize do not take cancellationTokens.